crates/runestick/src/any_obj.rs

```diff
@@ -168,7 +208,54 @@ impl AnyObj {
                 type_name: type_name_impl::<T>,
                 type_hash: type_hash_impl::<T>,
             },
-            data: data as *mut _ as *mut () as *const (),
+            data: data as *const _ as *const (),
+        }
+    }
+
+    /// Construct an Any that wraps a mutable pointer. This overload lets you
+    /// pass in a lock or other guard-type, as well as the raw value.
+    ///
+    /// # Safety
+    ///
+    /// Caller must ensure that the returned `AnyObj` doesn't outlive the
+    /// reference it is wrapping and the source of the look.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use runestick::{Any, Shared, AnyObj};
+    ///
+    /// #[derive(Any)]
+    /// struct Thing(u32);
+    ///
+    /// let mut value = std::cell::RefCell::new(Thing(10u32));
+    /// let guard = value.borrow_mut();
+    ///
+    /// unsafe {
+    ///     let shared = Shared::new(AnyObj::from_mut_deref::<Thing, _>(guard));
+    ///     shared.downcast_borrow_mut::<Thing>().unwrap().0 = 20;
+    ///     assert_eq!(20u32, shared.downcast_borrow_mut::<Thing>().unwrap().0);
+    ///     assert_eq!(20u32, shared.downcast_borrow_ref::<Thing>().unwrap().0);
+    ///     drop(shared);
+    ///     value.borrow();
+    /// }
+    /// ```
+    pub unsafe fn from_mut_deref<T, U: std::ops::DerefMut<Target = T>>(data: U) -> Self
```

Comment:
No harm in importing `DerefMut` into scope.
Also `from_deref_mut` to follow suggestion as above.